### ISSUE TITLE
Feat: Gradient accumulation

### ISSUE THREAD

**Issue by YoniSchirris** [lucidrains/byol-pytorch#28]
I've never worked with lightning, but the paper mentions that

_To avoid re-tuning other hyperparameters, we average gradients over N consecutive steps before updating the online network when reducing the batch size by a factor N . The target network is updated once every N steps, after the update of the online network; we accumulate the N -steps in parallel in our runs._

Which can be done using something like
```
# some code
# Initialize dataset with batch size 10
opt.zero_grad()
for i, (input, target) in enumerate(dataset):
    pred = net(input)
    loss = crit(pred, target)
    # one graph is created here
    loss.backward()
    # graph is cleared here
    if (i+1)%10 == 0:
        # every 10 iterations of batches of size 10
        opt.step()
        opt.zero_grad()
```
from https://discuss.pytorch.org/t/why-do-we-need-to-set-the-gradients-manually-to-zero-in-pytorch/4903/20

Might be useful to get maximum perfromance with your implementation

**Comment by lucidrains** [comment]
@YoniSchirris thanks for pointing this out! I've exposed the keyword argument in the latest commit https://github.com/lucidrains/byol-pytorch/commit/15deadd50195b1f16efc0e3fb027c3b6a555a117 so people can utilize this paper detail
